# Cleaning donor info 

In [23]:
import pandas as pd

Import the complete, raw data as I received it from the FAINT project team. It has 2057 rows and 155 columns. 

In [32]:
path_name = '/Users/dionnespaltman/Desktop/V6/FAINT_Info_Personality_timepoints.csv'
raw_donor_info = pd.read_csv(path_name)
display(raw_donor_info)

ID  Time_point  Gender   Age       Datum  Location  Condition  Weight  \
0       5         1.0       2  33.0  21-10-2019         0          1    74.0   
1       5         2.0       2  33.0  21-10-2019         0          1    74.0   
2       5         3.0       2  33.0  21-10-2019         0          1    74.0   
3       5         4.0       2  33.0  21-10-2019         0          1    74.0   
4       5         5.0       2  33.0  21-10-2019         0          1    74.0   
...   ...         ...     ...   ...         ...       ...        ...     ...   
2052  332         2.0       2  21.0  19-12-2022         3          2    66.0   
2053  332         4.0       2  21.0  19-12-2022         3          2    66.0   
2054  332         5.0       2  21.0  19-12-2022         3          2    66.0   
2055  332         6.0       2  21.0  19-12-2022         3          2    66.0   
2056  332         7.0       2  21.0  19-12-2022         3          2    66.0   

      Length  FEAR_sum  ...  Psychological_sum  Faintnessn  Dizzinessn  \
0      171.0        26  ...                8.0         1.0         1.0   
1      171.0        26  ...                6.0         1.0         1.0   
2      171.0        26  ...                6.0         1.0         1.0   
3      171.0        26  ...                5.0         1.0         1.0   
4      171.0        26  ...                5.0         1.0         1.0   
...      ...       ...  ...                ...         ...         ...   
2052   173.0        26  ...               12.0         1.0         1.0   
2053   173.0        26  ...               14.0         1.0         1.0   
2054   173.0        26  ...               12.0         1.0         1.0   
2055   173.0        26  ...                9.0         1.0         1.0   
2056   173.0        26  ...                6.0         1.0         1.0   

      Weaknessn  Lightheadednessn  Fearn  Tensionn  Stressn  Nervousnessn  \
0           1.0               1.0    2.0       2.0      2.0           2.0   
1           1.0               1.0    1.0       2.0      1.0           2.0   
2           1.0               1.0    1.0       2.0      1.0           2.0   
3           1.0               1.0    1.0       1.0      1.0           2.0   
4           1.0               1.0    1.0       1.0      1.0           2.0   
...         ...               ...    ...       ...      ...           ...   
2052        1.0               1.0    2.0       3.0      3.0           4.0   
2053        1.0               1.0    3.0       3.0      3.0           5.0   
2054        1.0               1.0    2.0       3.0      3.0           4.0   
2055        1.0               1.0    2.0       2.0      2.0           3.0   
2056        1.0               1.0    1.0       1.0      2.0           2.0   

      na.rm  
0      True  
1      True  
2      True  
3      True  
4      True  
...     ...  
2052   True  
2053   True  
2054   True  
2055   True  
2056   True  

[2057 rows x 155 columns]

# Unique IDs
Check how many unique IDs we have. We have 328. 

In [33]:
unique_id_count = raw_donor_info['ID'].nunique()
print("Number of unique IDs:", unique_id_count)

Number of unique IDs: 328


We will delete the 8 IDs that appear only once in the data. This means that they were only in the first stage. Now we have a dataframe with 2049 rows and 155 columns. 

In [35]:
# Group by 'ID' and count occurrences
id_counts = raw_donor_info.groupby('ID').size().reset_index(name='count')
print(id_counts)

      ID  count
0      5      7
1      6      7
2      7      7
3      8      7
4      9      7
..   ...    ...
323  328      6
324  329      6
325  330      6
326  331      6
327  332      6

[328 rows x 2 columns]


In [36]:
# Group by 'ID' and count occurrences
id_counts = raw_donor_info.groupby('ID').size().reset_index(name='count')

# Filter the IDs that appear only once
ids_once = id_counts[id_counts['count'] == 1]['ID']

# Print the list of IDs that appear only once
print("List of IDs that appear only once:")
print(ids_once.tolist())

List of IDs that appear only once:
[11, 20, 55, 56, 70, 89, 105, 110]


In [37]:
# Filter out these IDs from the original DataFrame
raw_donor_info = raw_donor_info[~raw_donor_info['ID'].isin(ids_once)]

# Optionally, if you want to see the distribution of counts excluding those with a count of 1
count_distribution = id_counts[id_counts['count'] > 1].groupby('count').size().reset_index(name='ID_count')

# Print the count distribution
print(count_distribution)

   count  ID_count
0      6       191
1      7       129


In [38]:
print(raw_donor_info['ID'].nunique())

320


Depending on the location, some participants didn't measure their VVR at stage 3. 

In [39]:
stage_counts = raw_donor_info['Time_point'].value_counts().reset_index()
stage_counts.columns = ['Timepoint', 'Occurrences']
timepoint_counts = stage_counts.sort_values(by='Timepoint')

print(timepoint_counts)

   Timepoint  Occurrences
1        1.0          320
2        2.0          320
6        3.0          128
3        4.0          320
4        5.0          320
0        6.0          321
5        7.0          320


# Dropping columns 
There are a lot of columns that are not interesting for the current research. So these will be dropped. 

In [40]:
columns = list(raw_donor_info.columns)
print(columns)

['ID', 'Time_point', 'Gender', 'Age', 'Datum', 'Location', 'Condition', 'Weight', 'Length', 'FEAR_sum', 'VVR_sum_tp', 'VVR_psych_tp', 'VVR_phys_tp', 'ERQ_sum', 'ERQ_CR', 'ERQ_ES', 'SSAS_sum', 'ASI_sum', 'MAIA_sum', 'MAIA_noticing_mean', 'MAIA_notdistr_mean', 'MAIA_notworry_mean', 'MAIA_attregul_mean', 'MAIA_emoaware_mean', 'MAIA_selfregu_mean', 'MAIA_bodylist_mean', 'MAIA_trust_mean', 'MAIA_noticing_sum', 'MAIA_notdistr_sum', 'MAIA_notworry_sum', 'MAIA_attregul_sum', 'MAIA_emoaware_sum', 'MAIA_selfregu_sum', 'MAIA_bodylist_sum', 'MAIA_trust_sum', 'Q2.2_1', 'Q2.2_2', 'Q2.2_3', 'Q2.2_4', 'Q2.2_5', 'Q2.2_6', 'Q2.2_7', 'Q2.2_8', 'Q2.2_9', 'Q2.2_10', 'Q3.1_1', 'Q3.1_2', 'Q3.1_3', 'Q3.1_4', 'Q3.1_5', 'Q3.1_6', 'Q3.1_7', 'Q3.1_8', 'Q3.2_1', 'Q3.2_2', 'Q3.2_3', 'Q3.2_4', 'Q3.2_5', 'Q3.2_6', 'Q3.2_7', 'Q3.2_8', 'Q4.1_1', 'Q4.1_2', 'Q4.1_3', 'Q4.1_4', 'Q4.1_5', 'Q4.1_6', 'Q4.1_7', 'Q4.1_8', 'Q4.1_9', 'Q4.1_10', 'Q5.1_1', 'Q5.1_2', 'Q5.1_3', 'Q5.1_4', 'Q5.1_5', 'Q5.1_6', 'Q5.1_7', 'Q5.1_8', 'Q5.1

In [41]:
# Columns to drop
columns_to_drop = ['FEAR_sum', 'VVR_sum_tp', 'VVR_psych_tp', 'VVR_phys_tp', 'ERQ_sum', 'ERQ_CR', 'ERQ_ES', 'SSAS_sum', 'ASI_sum', 'MAIA_sum', 
                   'MAIA_noticing_mean', 'MAIA_notdistr_mean', 'MAIA_notworry_mean', 'MAIA_attregul_mean', 'MAIA_emoaware_mean', 'MAIA_selfregu_mean', 
                   'MAIA_bodylist_mean', 'MAIA_trust_mean', 'MAIA_noticing_sum', 'MAIA_notdistr_sum', 'MAIA_notworry_sum', 'MAIA_attregul_sum',
                     'MAIA_emoaware_sum', 'MAIA_selfregu_sum', 'MAIA_bodylist_sum', 'MAIA_trust_sum', 'Q2.2_1', 'Q2.2_2', 'Q2.2_3', 'Q2.2_4', 
                     'Q2.2_5', 'Q2.2_6', 'Q2.2_7', 'Q2.2_8', 'Q2.2_9', 'Q2.2_10', 'Q3.1_1', 'Q3.1_2', 'Q3.1_3', 'Q3.1_4', 'Q3.1_5', 'Q3.1_6', 
                     'Q3.1_7', 'Q3.1_8', 'Q3.2_1', 'Q3.2_2', 'Q3.2_3', 'Q3.2_4', 'Q3.2_5', 'Q3.2_6', 'Q3.2_7', 'Q3.2_8', 'Q4.1_1', 'Q4.1_2', 
                     'Q4.1_3', 'Q4.1_4', 'Q4.1_5', 'Q4.1_6', 'Q4.1_7', 'Q4.1_8', 'Q4.1_9', 'Q4.1_10', 'Q5.1_1', 'Q5.1_2', 'Q5.1_3', 'Q5.1_4', 
                     'Q5.1_5', 'Q5.1_6', 'Q5.1_7', 'Q5.1_8', 'Q5.1_9', 'Q5.1_10', 'Q5.1_11', 'Q5.1_12', 'Q5.1_13', 'Q5.1_14', 'Q5.1_15', 'Q5.1_16', 
                     'Q5.1_17', 'Q5.1_18', 'Q5.1_19', 'Q5.1_20', 'Q5.1_21', 'Q5.1_22', 'Q5.1_23', 'Q5.1_24', 'Q5.1_25', 'Q5.1_26', 'Q5.1_27', 'Q5.1_28', 
                     'Q5.1_29', 'Q5.1_30', 'Q5.1_31', 'Q5.1_32', 'Q5.1_33', 'Q5.1_34', 'Q5.1_35', 'Q5.1_36', 'Q5.1_37', 'Q14_1', 'Q14_2', 'Q14_3', 'Q14_4', 
                     'Q14_5', 'Q14_6', 'Q14_12', 'Q16_2', 'Q14_33', 'Q14_44', 'Q14_55', 'Q14_66', 'Q7.1_1', 'Q7.1_2', 'Q7.1_3', 'Q7.1_4', 'Q7.1_5', 'Q7.1_6', 
                     'Q7.1_7', 'Q7.1_8', 'Q7.1_9', 'Q7.1_10', 'Q7.1_11', 'Q7.1_12', 'Q7.1_13', 'Q7.1_14', 'Q7.1_15', 'Q7.1_16', 'Faintnessn', 'Dizzinessn',
                       'Weaknessn', 'Lightheadednessn', 'Fearn', 'Tensionn', 'Stressn', 'Nervousnessn', 'na.rm']

# Drop the columns
raw_donor_info.drop(columns=columns_to_drop, inplace=True)

# Get list of the new columns 
columns = list(raw_donor_info.columns)
print(columns)

['ID', 'Time_point', 'Gender', 'Age', 'Datum', 'Location', 'Condition', 'Weight', 'Length', 'Faintness', 'Dizziness', 'Weakness', 'Lightheadedness', 'Fear', 'Tension', 'Stress', 'Nervousness', 'Physical_sum', 'Psychological_sum']


# Renaming columns 

In [ ]:
raw_donor_info.rename(columns={'Datum': 'Date'}, inplace=True)
raw_donor_info.rename(columns={'Time_point': 'Stage'}, inplace=True)
raw_donor_info['Date'] = pd.to_datetime(raw_donor_info['Date'])

In [43]:
columns = list(raw_donor_info.columns)
print(columns)

['ID', 'Stage', 'Gender', 'Age', 'Date', 'Location', 'Condition', 'Weight', 'Length', 'Faintness', 'Dizziness', 'Weakness', 'Lightheadedness', 'Fear', 'Tension', 'Stress', 'Nervousness', 'Physical_sum', 'Psychological_sum']


# Preprocessing donor info
First we load the data. Raw_donor_info has 2049 rows and 19 columns. 

In [44]:
display(raw_donor_info)

ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       5    1.0       2  33.0 2019-10-21         0          1    74.0   
1       5    2.0       2  33.0 2019-10-21         0          1    74.0   
2       5    3.0       2  33.0 2019-10-21         0          1    74.0   
3       5    4.0       2  33.0 2019-10-21         0          1    74.0   
4       5    5.0       2  33.0 2019-10-21         0          1    74.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2052  332    2.0       2  21.0 2022-12-19         3          2    66.0   
2053  332    4.0       2  21.0 2022-12-19         3          2    66.0   
2054  332    5.0       2  21.0 2022-12-19         3          2    66.0   
2055  332    6.0       2  21.0 2022-12-19         3          2    66.0   
2056  332    7.0       2  21.0 2022-12-19         3          2    66.0   

      Length  Faintness  Dizziness  Weakness  Lightheadedness  Fear  Tension  \
0      171.0        1.0        1.0       1.0              1.0   2.0      2.0   
1      171.0        1.0        1.0       1.0              1.0   1.0      2.0   
2      171.0        1.0        1.0       1.0              1.0   1.0      2.0   
3      171.0        1.0        1.0       1.0              1.0   1.0      1.0   
4      171.0        1.0        1.0       1.0              1.0   1.0      1.0   
...      ...        ...        ...       ...              ...   ...      ...   
2052   173.0        1.0        1.0       1.0              1.0   2.0      3.0   
2053   173.0        1.0        1.0       1.0              1.0   3.0      3.0   
2054   173.0        1.0        1.0       1.0              1.0   2.0      3.0   
2055   173.0        1.0        1.0       1.0              1.0   2.0      2.0   
2056   173.0        1.0        1.0       1.0              1.0   1.0      1.0   

      Stress  Nervousness  Physical_sum  Psychological_sum  
0        2.0          2.0           4.0                8.0  
1        1.0          2.0           4.0                6.0  
2        1.0          2.0           4.0                6.0  
3        1.0          2.0           4.0                5.0  
4        1.0          2.0           4.0                5.0  
...      ...          ...           ...                ...  
2052     3.0          4.0           4.0               12.0  
2053     3.0          5.0           4.0               14.0  
2054     3.0          4.0           4.0               12.0  
2055     2.0          3.0           4.0                9.0  
2056     2.0          2.0           4.0                6.0  

[2049 rows x 19 columns]

# Adding sum scores per stage
We'll make the sum scores. These are the VVR measurements, so the physical measurements (faintness, dizziness, weakness, lightheadedness) and emotional measurements (fear, tension, stress, nervousness). 

In [45]:
# Sum the specified measurements to create the new column
raw_donor_info['Sum_VVR'] = raw_donor_info[['Faintness', 'Dizziness', 'Weakness', 'Lightheadedness', 'Fear', 'Tension', 'Stress', 'Nervousness']].sum(axis=1, skipna=True)

# Get the maximum and minimum values of the sum_VVR column
max_sum_VVR = raw_donor_info['Sum_VVR'].max()
min_sum_VVR = raw_donor_info['Sum_VVR'].min()

print("Maximum value of sum_VVR:", max_sum_VVR)
print("Minimum value of sum_VVR:", min_sum_VVR)

display(raw_donor_info)

Maximum value of sum_VVR: 40.0
Minimum value of sum_VVR: 8.0


ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       5    1.0       2  33.0 2019-10-21         0          1    74.0   
1       5    2.0       2  33.0 2019-10-21         0          1    74.0   
2       5    3.0       2  33.0 2019-10-21         0          1    74.0   
3       5    4.0       2  33.0 2019-10-21         0          1    74.0   
4       5    5.0       2  33.0 2019-10-21         0          1    74.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2052  332    2.0       2  21.0 2022-12-19         3          2    66.0   
2053  332    4.0       2  21.0 2022-12-19         3          2    66.0   
2054  332    5.0       2  21.0 2022-12-19         3          2    66.0   
2055  332    6.0       2  21.0 2022-12-19         3          2    66.0   
2056  332    7.0       2  21.0 2022-12-19         3          2    66.0   

      Length  Faintness  Dizziness  Weakness  Lightheadedness  Fear  Tension  \
0      171.0        1.0        1.0       1.0              1.0   2.0      2.0   
1      171.0        1.0        1.0       1.0              1.0   1.0      2.0   
2      171.0        1.0        1.0       1.0              1.0   1.0      2.0   
3      171.0        1.0        1.0       1.0              1.0   1.0      1.0   
4      171.0        1.0        1.0       1.0              1.0   1.0      1.0   
...      ...        ...        ...       ...              ...   ...      ...   
2052   173.0        1.0        1.0       1.0              1.0   2.0      3.0   
2053   173.0        1.0        1.0       1.0              1.0   3.0      3.0   
2054   173.0        1.0        1.0       1.0              1.0   2.0      3.0   
2055   173.0        1.0        1.0       1.0              1.0   2.0      2.0   
2056   173.0        1.0        1.0       1.0              1.0   1.0      1.0   

      Stress  Nervousness  Physical_sum  Psychological_sum  Sum_VVR  
0        2.0          2.0           4.0                8.0     12.0  
1        1.0          2.0           4.0                6.0     10.0  
2        1.0          2.0           4.0                6.0     10.0  
3        1.0          2.0           4.0                5.0      9.0  
4        1.0          2.0           4.0                5.0      9.0  
...      ...          ...           ...                ...      ...  
2052     3.0          4.0           4.0               12.0     16.0  
2053     3.0          5.0           4.0               14.0     18.0  
2054     3.0          4.0           4.0               12.0     16.0  
2055     2.0          3.0           4.0                9.0     13.0  
2056     2.0          2.0           4.0                6.0     10.0  

[2049 rows x 20 columns]

In [46]:
# Filter rows where Stage is 1 or 2 and group by ID to calculate the sum of Sum_VVR
temp = raw_donor_info[raw_donor_info['Stage'].isin([1, 2])]
VVR_scores = temp.groupby('ID')['Sum_VVR'].sum().reset_index()

# Rename the column to Sum_12
VVR_scores.rename(columns={'Sum_VVR': 'Sum_12'}, inplace=True)

# Merge the sum_12 values back into the original DataFrame
raw_donor_info = raw_donor_info.merge(VVR_scores, on='ID', how='left')

# Fill NaN values in Sum_12 with 0 if necessary
raw_donor_info['Sum_12'] = raw_donor_info['Sum_12'].fillna(0)

# Get the maximum and minimum values of the Sum_12 column
max_sum_12 = raw_donor_info['Sum_12'].max()
min_sum_12 = raw_donor_info['Sum_12'].min()

print("Maximum value of Sum_12:", max_sum_12)
print("Minimum value of Sum_12:", min_sum_12)

# Display the updated raw_donor_info DataFrame
display(raw_donor_info)

Maximum value of Sum_12: 42.0
Minimum value of Sum_12: 16.0


ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       5    1.0       2  33.0 2019-10-21         0          1    74.0   
1       5    2.0       2  33.0 2019-10-21         0          1    74.0   
2       5    3.0       2  33.0 2019-10-21         0          1    74.0   
3       5    4.0       2  33.0 2019-10-21         0          1    74.0   
4       5    5.0       2  33.0 2019-10-21         0          1    74.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2044  332    2.0       2  21.0 2022-12-19         3          2    66.0   
2045  332    4.0       2  21.0 2022-12-19         3          2    66.0   
2046  332    5.0       2  21.0 2022-12-19         3          2    66.0   
2047  332    6.0       2  21.0 2022-12-19         3          2    66.0   
2048  332    7.0       2  21.0 2022-12-19         3          2    66.0   

      Length  Faintness  ...  Weakness  Lightheadedness  Fear  Tension  \
0      171.0        1.0  ...       1.0              1.0   2.0      2.0   
1      171.0        1.0  ...       1.0              1.0   1.0      2.0   
2      171.0        1.0  ...       1.0              1.0   1.0      2.0   
3      171.0        1.0  ...       1.0              1.0   1.0      1.0   
4      171.0        1.0  ...       1.0              1.0   1.0      1.0   
...      ...        ...  ...       ...              ...   ...      ...   
2044   173.0        1.0  ...       1.0              1.0   2.0      3.0   
2045   173.0        1.0  ...       1.0              1.0   3.0      3.0   
2046   173.0        1.0  ...       1.0              1.0   2.0      3.0   
2047   173.0        1.0  ...       1.0              1.0   2.0      2.0   
2048   173.0        1.0  ...       1.0              1.0   1.0      1.0   

      Stress  Nervousness  Physical_sum  Psychological_sum  Sum_VVR  Sum_12  
0        2.0          2.0           4.0                8.0     12.0    22.0  
1        1.0          2.0           4.0                6.0     10.0    22.0  
2        1.0          2.0           4.0                6.0     10.0    22.0  
3        1.0          2.0           4.0                5.0      9.0    22.0  
4        1.0          2.0           4.0                5.0      9.0    22.0  
...      ...          ...           ...                ...      ...     ...  
2044     3.0          4.0           4.0               12.0     16.0    30.0  
2045     3.0          5.0           4.0               14.0     18.0    30.0  
2046     3.0          4.0           4.0               12.0     16.0    30.0  
2047     2.0          3.0           4.0                9.0     13.0    30.0  
2048     2.0          2.0           4.0                6.0     10.0    30.0  

[2049 rows x 21 columns]

# Creating dependent variable: sum VVR stages 4, 5, 6

In [47]:
# Filter rows where Stage is 4, 5, or 6 and group by ID to calculate the sum of Sum_VVR
temp_456 = raw_donor_info[raw_donor_info['Stage'].isin([4, 5, 6])]
VVR_scores_456 = temp_456.groupby('ID')['Sum_VVR'].sum().reset_index()

# Rename the column to Sum_456
VVR_scores_456.rename(columns={'Sum_VVR': 'Sum_456'}, inplace=True)

# Merge the sum_456 values back into the original DataFrame
raw_donor_info = raw_donor_info.merge(VVR_scores_456, on='ID', how='left')

# Fill NaN values in Sum_456 with 0 if necessary
raw_donor_info['Sum_456'] = raw_donor_info['Sum_456'].fillna(0)

# Get the maximum and minimum values of the Sum_456 column
max_sum_456 = raw_donor_info['Sum_456'].max()
min_sum_456 = raw_donor_info['Sum_456'].min()

print("Maximum value of Sum_456:", max_sum_456)
print("Minimum value of Sum_456:", min_sum_456)

# Display the updated raw_donor_info DataFrame
display(raw_donor_info)


Maximum value of Sum_456: 69.0
Minimum value of Sum_456: 24.0


ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       5    1.0       2  33.0 2019-10-21         0          1    74.0   
1       5    2.0       2  33.0 2019-10-21         0          1    74.0   
2       5    3.0       2  33.0 2019-10-21         0          1    74.0   
3       5    4.0       2  33.0 2019-10-21         0          1    74.0   
4       5    5.0       2  33.0 2019-10-21         0          1    74.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2044  332    2.0       2  21.0 2022-12-19         3          2    66.0   
2045  332    4.0       2  21.0 2022-12-19         3          2    66.0   
2046  332    5.0       2  21.0 2022-12-19         3          2    66.0   
2047  332    6.0       2  21.0 2022-12-19         3          2    66.0   
2048  332    7.0       2  21.0 2022-12-19         3          2    66.0   

      Length  Faintness  ...  Lightheadedness  Fear  Tension  Stress  \
0      171.0        1.0  ...              1.0   2.0      2.0     2.0   
1      171.0        1.0  ...              1.0   1.0      2.0     1.0   
2      171.0        1.0  ...              1.0   1.0      2.0     1.0   
3      171.0        1.0  ...              1.0   1.0      1.0     1.0   
4      171.0        1.0  ...              1.0   1.0      1.0     1.0   
...      ...        ...  ...              ...   ...      ...     ...   
2044   173.0        1.0  ...              1.0   2.0      3.0     3.0   
2045   173.0        1.0  ...              1.0   3.0      3.0     3.0   
2046   173.0        1.0  ...              1.0   2.0      3.0     3.0   
2047   173.0        1.0  ...              1.0   2.0      2.0     2.0   
2048   173.0        1.0  ...              1.0   1.0      1.0     2.0   

      Nervousness  Physical_sum  Psychological_sum  Sum_VVR  Sum_12  Sum_456  
0             2.0           4.0                8.0     12.0    22.0     26.0  
1             2.0           4.0                6.0     10.0    22.0     26.0  
2             2.0           4.0                6.0     10.0    22.0     26.0  
3             2.0           4.0                5.0      9.0    22.0     26.0  
4             2.0           4.0                5.0      9.0    22.0     26.0  
...           ...           ...                ...      ...     ...      ...  
2044          4.0           4.0               12.0     16.0    30.0     47.0  
2045          5.0           4.0               14.0     18.0    30.0     47.0  
2046          4.0           4.0               12.0     16.0    30.0     47.0  
2047          3.0           4.0                9.0     13.0    30.0     47.0  
2048          2.0           4.0                6.0     10.0    30.0     47.0  

[2049 rows x 22 columns]

# Creating sum of stage 4, 5, 6, 7

In [48]:
# Filter rows where Stage is 4, 5, 6, or 7 and group by ID to calculate the sum of Sum_VVR
temp_4567 = raw_donor_info[raw_donor_info['Stage'].isin([4, 5, 6, 7])]
VVR_scores_4567 = temp_4567.groupby('ID')['Sum_VVR'].sum().reset_index()

# Rename the column to Sum_4567
VVR_scores_4567.rename(columns={'Sum_VVR': 'Sum_4567'}, inplace=True)

# Merge the sum_4567 values back into the original DataFrame
raw_donor_info = raw_donor_info.merge(VVR_scores_4567, on='ID', how='left')

# Fill NaN values in Sum_4567 with 0 if necessary
raw_donor_info['Sum_4567'] = raw_donor_info['Sum_4567'].fillna(0)

# Get the maximum and minimum values of the Sum_4567 column
max_sum_4567 = raw_donor_info['Sum_4567'].max()
min_sum_4567 = raw_donor_info['Sum_4567'].min()

print("Maximum value of Sum_4567:", max_sum_4567)
print("Minimum value of Sum_4567:", min_sum_4567)

# Display the updated raw_donor_info DataFrame
display(raw_donor_info)


Maximum value of Sum_4567: 82.0
Minimum value of Sum_4567: 32.0


ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       5    1.0       2  33.0 2019-10-21         0          1    74.0   
1       5    2.0       2  33.0 2019-10-21         0          1    74.0   
2       5    3.0       2  33.0 2019-10-21         0          1    74.0   
3       5    4.0       2  33.0 2019-10-21         0          1    74.0   
4       5    5.0       2  33.0 2019-10-21         0          1    74.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2044  332    2.0       2  21.0 2022-12-19         3          2    66.0   
2045  332    4.0       2  21.0 2022-12-19         3          2    66.0   
2046  332    5.0       2  21.0 2022-12-19         3          2    66.0   
2047  332    6.0       2  21.0 2022-12-19         3          2    66.0   
2048  332    7.0       2  21.0 2022-12-19         3          2    66.0   

      Length  Faintness  ...  Fear  Tension  Stress  Nervousness  \
0      171.0        1.0  ...   2.0      2.0     2.0          2.0   
1      171.0        1.0  ...   1.0      2.0     1.0          2.0   
2      171.0        1.0  ...   1.0      2.0     1.0          2.0   
3      171.0        1.0  ...   1.0      1.0     1.0          2.0   
4      171.0        1.0  ...   1.0      1.0     1.0          2.0   
...      ...        ...  ...   ...      ...     ...          ...   
2044   173.0        1.0  ...   2.0      3.0     3.0          4.0   
2045   173.0        1.0  ...   3.0      3.0     3.0          5.0   
2046   173.0        1.0  ...   2.0      3.0     3.0          4.0   
2047   173.0        1.0  ...   2.0      2.0     2.0          3.0   
2048   173.0        1.0  ...   1.0      1.0     2.0          2.0   

      Physical_sum  Psychological_sum  Sum_VVR  Sum_12  Sum_456  Sum_4567  
0              4.0                8.0     12.0    22.0     26.0      34.0  
1              4.0                6.0     10.0    22.0     26.0      34.0  
2              4.0                6.0     10.0    22.0     26.0      34.0  
3              4.0                5.0      9.0    22.0     26.0      34.0  
4              4.0                5.0      9.0    22.0     26.0      34.0  
...            ...                ...      ...     ...      ...       ...  
2044           4.0               12.0     16.0    30.0     47.0      57.0  
2045           4.0               14.0     18.0    30.0     47.0      57.0  
2046           4.0               12.0     16.0    30.0     47.0      57.0  
2047           4.0                9.0     13.0    30.0     47.0      57.0  
2048           4.0                6.0     10.0    30.0     47.0      57.0  

[2049 rows x 23 columns]

# Adding stage 1 and 2 separately 

In [49]:
# Filter rows where Stage is 1 and group by ID to calculate the sum of Sum_VVR
temp_1 = raw_donor_info[raw_donor_info['Stage'] == 1]
VVR_scores_1 = temp_1.groupby('ID')['Sum_VVR'].sum().reset_index()

# Rename the column to Sum_1
VVR_scores_1.rename(columns={'Sum_VVR': 'Sum_1'}, inplace=True)

# Merge the sum_1 values back into the original DataFrame
raw_donor_info = raw_donor_info.merge(VVR_scores_1, on='ID', how='left')

# Fill NaN values in Sum_1 with 0 if necessary
raw_donor_info['Sum_1'] = raw_donor_info['Sum_1'].fillna(0)

# Get the maximum and minimum values of the Sum_1 column
max_sum_1 = raw_donor_info['Sum_1'].max()
min_sum_1 = raw_donor_info['Sum_1'].min()

print("Maximum value of Sum_1:", max_sum_1)
print("Minimum value of Sum_1:", min_sum_1)

Maximum value of Sum_1: 21.0
Minimum value of Sum_1: 8.0


In [50]:
# Filter rows where Stage is 2 and group by ID to calculate the sum of Sum_VVR
temp_2 = raw_donor_info[raw_donor_info['Stage'] == 2]
VVR_scores_2 = temp_2.groupby('ID')['Sum_VVR'].sum().reset_index()

# Rename the column to Sum_2
VVR_scores_2.rename(columns={'Sum_VVR': 'Sum_2'}, inplace=True)

# Merge the sum_1 values back into the original DataFrame
raw_donor_info = raw_donor_info.merge(VVR_scores_2, on='ID', how='left')

# Fill NaN values in Sum_1 with 0 if necessary
raw_donor_info['Sum_2'] = raw_donor_info['Sum_2'].fillna(0)

# Get the maximum and minimum values of the Sum_1 column
max_sum_2 = raw_donor_info['Sum_2'].max()
min_sum_2 = raw_donor_info['Sum_2'].min()

print("Maximum value of Sum_2:", max_sum_2)
print("Minimum value of Sum_2:", min_sum_2)

Maximum value of Sum_2: 22.0
Minimum value of Sum_2: 8.0


In [51]:
display(raw_donor_info)

ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       5    1.0       2  33.0 2019-10-21         0          1    74.0   
1       5    2.0       2  33.0 2019-10-21         0          1    74.0   
2       5    3.0       2  33.0 2019-10-21         0          1    74.0   
3       5    4.0       2  33.0 2019-10-21         0          1    74.0   
4       5    5.0       2  33.0 2019-10-21         0          1    74.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2044  332    2.0       2  21.0 2022-12-19         3          2    66.0   
2045  332    4.0       2  21.0 2022-12-19         3          2    66.0   
2046  332    5.0       2  21.0 2022-12-19         3          2    66.0   
2047  332    6.0       2  21.0 2022-12-19         3          2    66.0   
2048  332    7.0       2  21.0 2022-12-19         3          2    66.0   

      Length  Faintness  ...  Stress  Nervousness  Physical_sum  \
0      171.0        1.0  ...     2.0          2.0           4.0   
1      171.0        1.0  ...     1.0          2.0           4.0   
2      171.0        1.0  ...     1.0          2.0           4.0   
3      171.0        1.0  ...     1.0          2.0           4.0   
4      171.0        1.0  ...     1.0          2.0           4.0   
...      ...        ...  ...     ...          ...           ...   
2044   173.0        1.0  ...     3.0          4.0           4.0   
2045   173.0        1.0  ...     3.0          5.0           4.0   
2046   173.0        1.0  ...     3.0          4.0           4.0   
2047   173.0        1.0  ...     2.0          3.0           4.0   
2048   173.0        1.0  ...     2.0          2.0           4.0   

      Psychological_sum  Sum_VVR  Sum_12  Sum_456  Sum_4567  Sum_1  Sum_2  
0                   8.0     12.0    22.0     26.0      34.0   12.0   10.0  
1                   6.0     10.0    22.0     26.0      34.0   12.0   10.0  
2                   6.0     10.0    22.0     26.0      34.0   12.0   10.0  
3                   5.0      9.0    22.0     26.0      34.0   12.0   10.0  
4                   5.0      9.0    22.0     26.0      34.0   12.0   10.0  
...                 ...      ...     ...      ...       ...    ...    ...  
2044               12.0     16.0    30.0     47.0      57.0   14.0   16.0  
2045               14.0     18.0    30.0     47.0      57.0   14.0   16.0  
2046               12.0     16.0    30.0     47.0      57.0   14.0   16.0  
2047                9.0     13.0    30.0     47.0      57.0   14.0   16.0  
2048                6.0     10.0    30.0     47.0      57.0   14.0   16.0  

[2049 rows x 25 columns]

# Filtering raw_donor_df with the right IDs

ID list of June 3, 2024: 

[ 80  87  78  92  38 129  95 118  49  94  39  93  79  81  48 119 114  57
 146 113  50  45 106  42 101  74 130  73  99  66  33 125  61  34 122  43
 100  44 107 112  51 115 140  35  60 123  32  98 124 131  72 136  75  46
  41 102 142 117  54 145  53  65  30  62  37 121 134  77  88 133  67 111
 144  52 116 143  40 103  47 104 132  71 135  76  36  63 120  31  64 127
  91  69  96  29  83  84 138  58  85 139  82  28  97  68  59 108  25  23
  24  26  27]

ID list of June 11, 2024:

[100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 23, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 24, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 25, 260, 261, 262, 26, 278, 279, 27, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 28, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 29, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 30, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 31, 321, 322, 323, 325, 329, 32, 330, 331, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 7, 80, 81, 82, 83, 84, 85, 87, 88, 89, 91, 92, 93, 94, 95, 96, 97, 98, 99]

In [52]:
id_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 23, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 24, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 25, 260, 261, 262, 26, 278, 279, 27, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 28, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 29, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 30, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 31, 321, 322, 323, 325, 329, 32, 330, 331, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 7, 80, 81, 82, 83, 84, 85, 87, 88, 89, 91, 92, 93, 94, 95, 96, 97, 98, 99]
print(len(id_list))

283


In [53]:
# Filter the DataFrame to include only rows where 'ID' is in id_list
filtered_donor_info = raw_donor_info[raw_donor_info['ID'].isin(id_list)]

# Display the filtered DataFrame
print("Filtered DataFrame with IDs in id_list:")
display(filtered_donor_info)

Filtered DataFrame with IDs in id_list:


ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
14      7    1.0       2  25.0 2019-10-28         0          2    71.0   
15      7    2.0       2  25.0 2019-10-28         0          2    71.0   
16      7    3.0       2  25.0 2019-10-28         0          2    71.0   
17      7    4.0       2  25.0 2019-10-28         0          2    71.0   
18      7    5.0       2  25.0 2019-10-28         0          2    71.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
2038  331    2.0       2  56.0 2022-11-30         0          3    73.0   
2039  331    4.0       2  56.0 2022-11-30         0          3    73.0   
2040  331    5.0       2  56.0 2022-11-30         0          3    73.0   
2041  331    6.0       2  56.0 2022-11-30         0          3    73.0   
2042  331    7.0       2  56.0 2022-11-30         0          3    73.0   

      Length  Faintness  ...  Stress  Nervousness  Physical_sum  \
14     172.0        1.0  ...     1.0          2.0           5.0   
15     172.0        1.0  ...     1.0          3.0           4.0   
16     172.0        1.0  ...     2.0          2.0           4.0   
17     172.0        2.0  ...     1.0          2.0           5.0   
18     172.0        1.0  ...     1.0          4.0           5.0   
...      ...        ...  ...     ...          ...           ...   
2038   167.0        1.0  ...     2.0          4.0           4.0   
2039   167.0        1.0  ...     2.0          2.0           4.0   
2040   167.0        1.0  ...     1.0          1.0           6.0   
2041   167.0        1.0  ...     1.0          1.0           4.0   
2042   167.0        1.0  ...     1.0          1.0           4.0   

      Psychological_sum  Sum_VVR  Sum_12  Sum_456  Sum_4567  Sum_1  Sum_2  
14                  6.0     11.0    23.0     65.0      77.0   11.0   12.0  
15                  8.0     12.0    23.0     65.0      77.0   11.0   12.0  
16                  7.0     11.0    23.0     65.0      77.0   11.0   12.0  
17                  6.0     11.0    23.0     65.0      77.0   11.0   12.0  
18                  9.0     14.0    23.0     65.0      77.0   11.0   12.0  
...                 ...      ...     ...      ...       ...    ...    ...  
2038               10.0     14.0    28.0     30.0      38.0   14.0   14.0  
2039                8.0     12.0    28.0     30.0      38.0   14.0   14.0  
2040                4.0     10.0    28.0     30.0      38.0   14.0   14.0  
2041                4.0      8.0    28.0     30.0      38.0   14.0   14.0  
2042                4.0      8.0    28.0     30.0      38.0   14.0   14.0  

[1779 rows x 25 columns]

In [54]:
# Get the list of unique IDs in the filtered DataFrame
unique_ids = filtered_donor_info['ID'].unique()

# Print the list of unique IDs
print("List of unique IDs in filtered DataFrame:")
print(unique_ids)

# Print the number of unique IDs
print("Number of unique IDs:", len(unique_ids))

List of unique IDs in filtered DataFrame:
[  7  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39
  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  57  58  59
  60  61  62  63  64  65  66  67  68  69  71  72  73  74  75  76  77  78
  79  80  81  82  83  84  85  87  88  91  92  93  94  95  96  97  98  99
 100 101 102 103 104 106 107 108 109 111 112 113 114 115 116 117 118 119
 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137
 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155
 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173
 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191
 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209
 210 211 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231
 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249
 250 251 252 253 254 255 256 257 258 259 260 261 262 278 279 280 281 282
 283 284 

# We end up with 278 unique IDs 

# Mean and creation of low and high VVR group

June 3, 2024: Mean of 'Sum_4567' column: 37.61267605633803

June 11, 2024: Mean of 'Sum_4567' column: 39.770657672849914

In [55]:
# Calculate the mean of the 'Sum_4567' column, excluding NaN values
mean_sum_4567 = filtered_donor_info['Sum_4567'].mean(skipna=True)

# Print the mean value
print("Mean of 'Sum_4567' column:", mean_sum_4567)

# Determine the group for each unique ID based on the mean of Sum_4567
id_group = filtered_donor_info.groupby('ID')['Sum_4567'].mean().apply(lambda x: 0 if x < mean_sum_4567 else 1).reset_index()

# Rename the column to 'VVR_group'
id_group.rename(columns={'Sum_4567': 'VVR_group'}, inplace=True)

# Merge the group information back into the original DataFrame
filtered_donor_info = filtered_donor_info.merge(id_group, on='ID', how='left')

# Display the updated DataFrame with VVR_group
display(filtered_donor_info)

Mean of 'Sum_4567' column: 39.770657672849914


ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       7    1.0       2  25.0 2019-10-28         0          2    71.0   
1       7    2.0       2  25.0 2019-10-28         0          2    71.0   
2       7    3.0       2  25.0 2019-10-28         0          2    71.0   
3       7    4.0       2  25.0 2019-10-28         0          2    71.0   
4       7    5.0       2  25.0 2019-10-28         0          2    71.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
1774  331    2.0       2  56.0 2022-11-30         0          3    73.0   
1775  331    4.0       2  56.0 2022-11-30         0          3    73.0   
1776  331    5.0       2  56.0 2022-11-30         0          3    73.0   
1777  331    6.0       2  56.0 2022-11-30         0          3    73.0   
1778  331    7.0       2  56.0 2022-11-30         0          3    73.0   

      Length  Faintness  ...  Nervousness  Physical_sum  Psychological_sum  \
0      172.0        1.0  ...          2.0           5.0                6.0   
1      172.0        1.0  ...          3.0           4.0                8.0   
2      172.0        1.0  ...          2.0           4.0                7.0   
3      172.0        2.0  ...          2.0           5.0                6.0   
4      172.0        1.0  ...          4.0           5.0                9.0   
...      ...        ...  ...          ...           ...                ...   
1774   167.0        1.0  ...          4.0           4.0               10.0   
1775   167.0        1.0  ...          2.0           4.0                8.0   
1776   167.0        1.0  ...          1.0           6.0                4.0   
1777   167.0        1.0  ...          1.0           4.0                4.0   
1778   167.0        1.0  ...          1.0           4.0                4.0   

      Sum_VVR  Sum_12  Sum_456  Sum_4567  Sum_1  Sum_2  VVR_group  
0        11.0    23.0     65.0      77.0   11.0   12.0          1  
1        12.0    23.0     65.0      77.0   11.0   12.0          1  
2        11.0    23.0     65.0      77.0   11.0   12.0          1  
3        11.0    23.0     65.0      77.0   11.0   12.0          1  
4        14.0    23.0     65.0      77.0   11.0   12.0          1  
...       ...     ...      ...       ...    ...    ...        ...  
1774     14.0    28.0     30.0      38.0   14.0   14.0          0  
1775     12.0    28.0     30.0      38.0   14.0   14.0          0  
1776     10.0    28.0     30.0      38.0   14.0   14.0          0  
1777      8.0    28.0     30.0      38.0   14.0   14.0          0  
1778      8.0    28.0     30.0      38.0   14.0   14.0          0  

[1779 rows x 26 columns]

June 3, 2024: 

Count of unique IDs in each 'VVR_group' column:
- 0  -  74
- 1  - 37

June 11, 2024: 
- 0   -  186
- 1   -  92

In [56]:
# Count the occurrences of each unique value in the 'VVR_group' column
vvr_group_counts = filtered_donor_info[['ID', 'VVR_group']].drop_duplicates()['VVR_group'].value_counts()

# Print the counts
print("Count of unique IDs in each 'VVR_group' column:")
print(vvr_group_counts)

Count of unique IDs in each 'VVR_group' column:
0    186
1     92
Name: VVR_group, dtype: int64


# Saving 
Shape of filtered_donor_info is 1779 rows × 26 columns. 

In [57]:
filtered_donor_info.to_csv('/Users/dionnespaltman/Desktop/V6/filtered_donor_info_11-06-2024.csv', sep=',')

In [58]:
display(filtered_donor_info)

ID  Stage  Gender   Age       Date  Location  Condition  Weight  \
0       7    1.0       2  25.0 2019-10-28         0          2    71.0   
1       7    2.0       2  25.0 2019-10-28         0          2    71.0   
2       7    3.0       2  25.0 2019-10-28         0          2    71.0   
3       7    4.0       2  25.0 2019-10-28         0          2    71.0   
4       7    5.0       2  25.0 2019-10-28         0          2    71.0   
...   ...    ...     ...   ...        ...       ...        ...     ...   
1774  331    2.0       2  56.0 2022-11-30         0          3    73.0   
1775  331    4.0       2  56.0 2022-11-30         0          3    73.0   
1776  331    5.0       2  56.0 2022-11-30         0          3    73.0   
1777  331    6.0       2  56.0 2022-11-30         0          3    73.0   
1778  331    7.0       2  56.0 2022-11-30         0          3    73.0   

      Length  Faintness  ...  Nervousness  Physical_sum  Psychological_sum  \
0      172.0        1.0  ...          2.0           5.0                6.0   
1      172.0        1.0  ...          3.0           4.0                8.0   
2      172.0        1.0  ...          2.0           4.0                7.0   
3      172.0        2.0  ...          2.0           5.0                6.0   
4      172.0        1.0  ...          4.0           5.0                9.0   
...      ...        ...  ...          ...           ...                ...   
1774   167.0        1.0  ...          4.0           4.0               10.0   
1775   167.0        1.0  ...          2.0           4.0                8.0   
1776   167.0        1.0  ...          1.0           6.0                4.0   
1777   167.0        1.0  ...          1.0           4.0                4.0   
1778   167.0        1.0  ...          1.0           4.0                4.0   

      Sum_VVR  Sum_12  Sum_456  Sum_4567  Sum_1  Sum_2  VVR_group  
0        11.0    23.0     65.0      77.0   11.0   12.0          1  
1        12.0    23.0     65.0      77.0   11.0   12.0          1  
2        11.0    23.0     65.0      77.0   11.0   12.0          1  
3        11.0    23.0     65.0      77.0   11.0   12.0          1  
4        14.0    23.0     65.0      77.0   11.0   12.0          1  
...       ...     ...      ...       ...    ...    ...        ...  
1774     14.0    28.0     30.0      38.0   14.0   14.0          0  
1775     12.0    28.0     30.0      38.0   14.0   14.0          0  
1776     10.0    28.0     30.0      38.0   14.0   14.0          0  
1777      8.0    28.0     30.0      38.0   14.0   14.0          0  
1778      8.0    28.0     30.0      38.0   14.0   14.0          0  

[1779 rows x 26 columns]